pkg/service/controller.go

```diff
@@ -36,72 +37,134 @@ var ControllerCaps = []csi.ControllerServiceCapability_RPC_Type{
+		if err != nil {
+			msg := fmt.Errorf("failed creating disk %s error is %w", diskName, err)
+			klog.Errorf(msg.Error())
+			return nil, msg
+		}
+		disk = createDisk.MustDisk()
+		finished, err := checkJobFinished(ctx, conn, correlationID)
+		if err != nil {
+			msg := fmt.Errorf("error while waiting for disk %s create job to finish, error: %w",
+				disk.MustId(), err)
+			klog.Errorf(msg.Error())
+			return nil, msg
+		}
+		if !finished {
+			msg := fmt.Errorf("create disk %s job didn't finish, error: %w", disk.MustId(), err)
+			klog.Errorf(msg.Error())
+			return nil, msg
+		}
 	}
-
-	// TODO rgolan the default in case of error would be non thin - change it?
-	thinProvisioning, _ := strconv.ParseBool(req.Parameters[ParameterThinProvisioning])
-
-	provisionedSize := req.CapacityRange.GetRequiredBytes()
-	if provisionedSize < minimumDiskSize {
-		provisionedSize = minimumDiskSize
-	}
-
-	// creating the disk
-	disk, err := ovirtsdk.NewDiskBuilder().
-		Name(req.Name).
-		StorageDomainsBuilderOfAny(*ovirtsdk.NewStorageDomainBuilder().Name(req.Parameters[ParameterStorageDomainName])).
-		ProvisionedSize(provisionedSize).
-		ReadOnly(false).
-		Format(ovirtsdk.DISKFORMAT_COW).
-		Sparse(thinProvisioning).
-		Build()
-
-	if err != nil {
-		// failed to construct the disk
-		return nil, err
-	}
-
-	createDisk, err := conn.SystemService().DisksService().
-		Add().
-		Disk(disk).
-		Send()
-	if err != nil {
-		// failed to create the disk
-		klog.Errorf("Failed creating disk %s", req.Name)
-		return nil, err
+	if err = waitForDiskStatusOk(ctx, conn, disk.MustId()); err != nil {
+		msg := fmt.Errorf("failed waiting for disk %s to be created, error: %w",
+			disk.MustId(), err)
+		klog.Errorf(msg.Error())
+		return nil, msg
 	}
+	klog.Infof("Finished creating disk %s", diskName)
 	return &csi.CreateVolumeResponse{
 		Volume: &csi.Volume{
-			CapacityBytes: createDisk.MustDisk().MustProvisionedSize(),
-			VolumeId:      createDisk.MustDisk().MustId(),
+			CapacityBytes: disk.MustProvisionedSize(),
+			VolumeId:      disk.MustId(),
 		},
 	}, nil
 }
 
+func handleCreateVolumeImageFormat(conn *ovirtsdk.Connection, storageDomainName string, thinProvisioning bool) (ovirtsdk.DiskFormat, error) {
+	sd, err := getStorageDomainByName(conn, storageDomainName)
+	if err != nil {
+		return "", fmt.Errorf(
+			"failed searching for storage domain with name %s, error: %w", storageDomainName, err)
+	}
+	if sd == nil {
+		return "", fmt.Errorf(
+			"storage domain with name %s wasn't found", storageDomainName)
+	}
+	storage, ok := sd.Storage()
+	if !ok {
+		return "", fmt.Errorf(
+			"storage domain with name %s didn't have host storage, veify it is connected to a host",
+			storageDomainName)
+	}
+	storageType, ok := storage.Type()
+	if !ok {
+		return "", fmt.Errorf(
+			"storage domain with name %s didn't have a storage type, please check storage domain on ovirt engine",
+			storageDomainName)
+	}
+	if storageType != ovirtsdk.STORAGETYPE_NFS && thinProvisioning {
```

Comment:
looks like we're gonna need a helper `isFileDomain(storageType)`, this doesn't cover gluster and posixfs

Reply:
What should we do about gluster and posixfs?
We talked about the following combinations(from [docs](https://access.redhat.com/documentation/en-us/red_hat_virtualization/4.4/html/administration_guide/understanding_virtual_disks)):
- NFSx+Preallocated=Raw
- SAN+Preallocated = Raw
- SAN+Sparse = QCOW2
- NFS+Sparse = Raw
What should we do for:
- gluster+Preallocated=?
- gluster+Sparse=?
- posixfs+Preallocated=?
- posixfs+Sparse=?

Reply:
all FS based domains behave the same